### DISCUSSION TITLE
What'll be your first Lowdefy project?

### DISCUSSION THREAD

**Discussion by DLvalentine** [lowdefy/lowdefy#469] [General]
Hey all! 

I just stumbled upon this after reading the Hacker News post about it - it came at just the right time, too. I am working on a little side project to learn Spring Boot, and wanted to build a little front-end to go along with it. Super simple REST API for CRUD operations, so nothing special - just learning the ropes!

I *was* going to pick a front-end framework/library and build it like I normally would, since I build front-ends by trade. But I thought that to rapidly prototype and build out a front-end for my project, that this seemed like a *great* fit. 

That said, I'm really happy this exists and want to see what people plan to use this for!

**Comment by astrogewgaw** [comment]
I found about **lowdefy** recently, and right when I was actually looking for something that will make building a web app easier for someone like me 😁 (who does not have any idea about web apps)! I am a research student, and I am looking into building a web app that will display a list of all scientists currently working in the field of [_astrochemistry_](https://en.wikipedia.org/wiki/Astrochemistry). To ensure that this list can be updated easily, I decided to maintain the original list in a GitHub repository. The idea is for users to suggest edits to the list through the web app. The edit should then be sent over to the GitHub repository where the original list is stored, which will trigger a bot. The bot will convert the edit into a pull request. I basically want users to be able to submit their edits to this list as pull requests to the GitHub repository where the original list is stored. I can then go over the pull requests, and merge/reject them as required. I think a GitHub connection would be pretty helpful (which is why I voted for it 😄). If anyone has ideas that could help me pull this off, it would be highly appreciated 😁 !

**Reply by SamTolmay** [reply]
Hi Ujjwal

In what format are you planning on storing the list?

I think the easiest would be to maintain the Lowdefy app in the same Github repository as the list of scientists. If the list is in YAML or JSON, you can easily reference the list in your configuration.

If you deploy the app with Netlify, you will also be able to preview Pull Requests and the live site will be updated when you merge them.

Please let me know if you need any help setting this up.

**Reply by astrogewgaw** [reply]
hey @SamTolmay! ya i am thinking of maintaining the list in JSON right now. also i already have a site up on Netlify, so no worries there as well :grin:! is there a way to change the base path of the app? as in, if i already have a site at, say, https://example.com, then my **lowdefy** app could be at https://example.com/app/? i would ideally like the lowdefy app to be hosted as a part of my site. but i am open to setting up a separate site as well. and yes, i will definitely let you know if i need any help (heads up, i probably will 😆)!

**Reply by Gervwyk** [reply]
Hi @astrogewgaw! So great to hear that you already have a deployment up. 

It will be the easiest to host the Lowdefy app on a subdomain, like https://app.example.com. Netlify has instructions on how to set up your DNS for that, but just shout if you get stuck.

For example, we are running lowdefy.com as a gatsby site and docs.lowdefy.com as a lowdefy site from two different repositories, this should also be doable in the same repository I think. 

For now, we try to keep the app router basic to keep things simple, but we are open to suggestions on how it can be improved.

**Reply by astrogewgaw** [reply]
hey @Gervwyk! while a sub-domain is not a bad idea, i think a base path change is better suited for what i have in mind (something like [this](https://nextjs.org/docs/migrating/incremental-adoption); see the **subpath** section). i have no idea how hard or easy this would be to implement. i leave the choice up to y'all who are maintaining this project for us :grin:!

**Comment by xuoguoto** [comment]
Testing this out for a simple dashboard. Lack of user authentication is a bummer for now, but I guess that will get solved soon.

**Reply by Gervwyk** [reply]
Seems like the only thing holding us back. Really excited to roll it out asap!

![image](https://user-images.githubusercontent.com/7165064/109549093-278bde80-7ad6-11eb-811b-62c5c3d0c3ec.gif)

**Comment by kurzai** [comment]
Probably a crm, since I dont find that most crms out there fit my needs :) can't wait to dive into the lowdefy code!

**Reply by Gervwyk** [reply]
Super! Great to hear. We've had great success with Lowdefy and CRM apps for customers, will probably create a community CRM example in the months to come. 

Have you looked into what data storage you will be using for your CRM?

**Reply by kurzai** [reply]
I currently store a lot of lead information in a managed postgresql instance but since that's not available yet in lowdefy to my knowledge I might just wait for the feature or try to contribute to it myself

**Reply by Gervwyk** [reply]
PostgreSQL will be the next connection we'll add, along with support for other SQL databases since we've had quite a few requests for this. It's likely that we'll implement the connector using [Knex](https://www.npmjs.com/package/knex) package. We're hoping to ship this connector in the first half of April.

In the mean time, you can get really far by building out the UI, and then just swapping the connector when PostgreSQL is available.

Also, any contributions are welcome!

**Reply by kurzai** [reply]
yeah man that's probably what I'm gonna do! :)
I'll probably post this in the requested features section aswell, but were there any ideas about kanban style blocks yet? I know it's basically just a horizontal row of lists, but the aspect of grouping a list of objects by a property to generate a kanban board and be able to drag and drop them between columns would be very convenient.

**Reply by Gervwyk** [reply]
Yes. Thats a really good suggestion! We spoke about it a while back, and did not take it up since. 

Please post on [new blocks](https://github.com/lowdefy/lowdefy/discussions/492), will be helpful to have all the ideas in one place. Also a few ideas regarding blocks like kanban, timeline, table and calendar which I would like to unpack.

**Comment by vaddisrinivas** [comment]
Having built a pretty basic sort of -  GoogleSheets-based CMS - [WebSheets](https://github.com/vaddisrinivas/WebSheets) a year ago, I now want to build a portfolio website with Lowdefy! 

Can't wait for the v4.0!

**Reply by Gervwyk** [reply]
Sweet! Your project looks really cool! I'll also be building the new Lowdefy landing page with Lowdefy pretty soon 🌈
